apps/contracts/vault/src/report.rs

```diff
@@ -51,9 +51,13 @@ impl Report {
 }
 
 pub fn distribute_strategy_fees(e: &Env, strategy_address: &Address, access_control: &AccessControl) -> Result<i128, ContractError> {
-    let report = get_report(&e, &strategy_address);
+    let mut report = get_report(&e, &strategy_address);
```

Comment:
This function should only distribute_strategy_fees, if we need to report and lock in other functions  we implement that, or maybe add another function that do the previous two steps.

for example one might want to only distribute the current fees since there is a loss from that moment, etc.
```suggestion
    let mut report = get_report(&e, &strategy_address);
```